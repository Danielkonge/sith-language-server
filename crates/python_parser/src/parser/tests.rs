mod context;
mod function;
mod invalid_assignments;
mod parser;
mod suite;
