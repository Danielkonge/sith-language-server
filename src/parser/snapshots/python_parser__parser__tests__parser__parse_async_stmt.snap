---
source: src/parser/tests.rs
expression: "parse(\"\nasync def f():\n    ...\n\nasync for i in iter:\n    ...\n\nasync with x:\n    ...\n\n@a\nasync def x():\n    ...\n\")"
---
ParsedFile {
    ast: Module {
        body: [
            AsyncFunctionDef(
                FunctionDefStmt {
                    name: Valid(
                        Identifier {
                            id: "f",
                            range: 11..12,
                        },
                    ),
                    parameters: Parameters {
                        kwarg: None,
                        vararg: None,
                        args: [],
                        posonlyargs: [],
                        kwonlyargs: [],
                        range: 13..14,
                    },
                    body: [
                        Expression(
                            ExpressionStmt {
                                value: Ellipsis(
                                    EllipsisExpr {
                                        range: 20..23,
                                    },
                                ),
                                range: 20..23,
                            },
                        ),
                    ],
                    decorators: [],
                    returns: None,
                    range: 1..23,
                },
            ),
            AsyncFor(
                ForStmt {
                    target: Id(
                        Identifier {
                            id: "i",
                            range: 35..36,
                        },
                    ),
                    iter: Id(
                        Identifier {
                            id: "iter",
                            range: 40..44,
                        },
                    ),
                    body: [
                        Expression(
                            ExpressionStmt {
                                value: Ellipsis(
                                    EllipsisExpr {
                                        range: 50..53,
                                    },
                                ),
                                range: 50..53,
                            },
                        ),
                    ],
                    orelse: [],
                    range: 25..53,
                },
            ),
            AsyncWith(
                WithStmt {
                    items: [
                        WithItem {
                            item: Id(
                                Identifier {
                                    id: "x",
                                    range: 66..67,
                                },
                            ),
                            target: None,
                            range: 66..67,
                        },
                    ],
                    body: [
                        Expression(
                            ExpressionStmt {
                                value: Ellipsis(
                                    EllipsisExpr {
                                        range: 73..76,
                                    },
                                ),
                                range: 73..76,
                            },
                        ),
                    ],
                    range: 55..76,
                },
            ),
            AsyncFunctionDef(
                FunctionDefStmt {
                    name: Valid(
                        Identifier {
                            id: "x",
                            range: 91..92,
                        },
                    ),
                    parameters: Parameters {
                        kwarg: None,
                        vararg: None,
                        args: [],
                        posonlyargs: [],
                        kwonlyargs: [],
                        range: 93..94,
                    },
                    body: [
                        Expression(
                            ExpressionStmt {
                                value: Ellipsis(
                                    EllipsisExpr {
                                        range: 100..103,
                                    },
                                ),
                                range: 100..103,
                            },
                        ),
                    ],
                    decorators: [
                        Decorator {
                            expr: Id(
                                Identifier {
                                    id: "a",
                                    range: 79..80,
                                },
                            ),
                            range: 78..80,
                        },
                    ],
                    returns: None,
                    range: 78..103,
                },
            ),
        ],
        range: 0..103,
    },
    parse_errors: [],
}
